src/directive.js

```diff
@@ -0,0 +1,83 @@
+import interpolate from './interpolate'
+import translate from './translate'
+import Vue from 'vue'
+
+// we have to cache the current language locally
+// to detect if the language has changed and translation should be updated
+let currentLanguage = null
```

Comment:
It's not really clear why you have to cache it here, can you elaborate a little more?